2024.1/serialization/serialization.adoc

```diff
@@ -0,0 +1,664 @@
+For fields of type <<string>>, <<boolean>>, <<integer>>, and <<literal>>, the _value_ is encoded as JSON _string_ in the same way as for a property.
+For fields of type <<structuredDatatype>>, the value is encoded as JSON _object_.
+
+.Structured datatypes used in the examples
+--
+
+NOTE: The format used in this example is non-normative.
+
+[source]
+----
+enumeration Currency                   [id aa, key currency]
+  literal EUR                          [id a0, key cur-eur]
+  literal GBP                          [id a1, key cur-gbp]
+
+structured datatype Amount             [id 10, key amount]
+  value: Integer                       [id 11, key amount-val]
+  currency: Currency                   [id 12, key amount-cur]
+  digital: Boolean                     [id 13, key digital]
+
+structured datatype Decimal            [id 20, key decimal]
+  int: Integer                         [id 21, key decimal-int]
+  frac: Integer                        [id 22, key decimal-frac]
+
+structured datatype ComplexNumber      [id 30, key complex]
+  real: Decimal                        [id 31, key complex-real]
+  imaginary: Decimal                   [id 32, key complex-imaginary]
+----
+--
+
+.Valid examples
+* Amount _42 EUR non-digital_: `"{\n  \"amount-val\":  \"42\",\n  \"amount-cur\": \"cur-eur\",\n  \"digital\": \"false\"\n}"`
++
+unescaped content:
++
+[source, json]
+----
+{
+  "amount-val":  "42",
+  "amount-cur": "cur-eur",
+  "digital": "false"
+}
+----
+
+* Decimal _42.0_: `"{\"decimal-int\": \"42\", \"decimal-frac\": \"0\"}"`
++
+unescaped content:
++
+[source, json]
+----
+{"decimal-int": "42", "decimal-frac": "0"}
+----
+
+* ComplexNumber _23.17 + 42.0i_: `"{\n\"complex-real\": { \"decimal-int\": \"23\", \"decimal-frac\": \"17\"},\n\"complex-imaginary\": { \"decimal-int\": \"42\", \"decimal-frac\": \"0\"}\n}"`
++
+unescaped content:
++
+[source, json]
+----
+{
+"complex-real": { "decimal-int": "23", "decimal-frac": "17"},
+"complex-imaginary": { "decimal-int": "42", "decimal-frac": "0"}
+}
+----
+
+.Invalid examples
+* Amount with non-string field values: `"{\n  \"amount-val\":  42,\n  \"amount-cur\": \"cur-eur\",\n  \"digital\": false\n}"`
++
+unescaped content:
++
+[source, json]
+----
+{
+  "amount-val":  42,
+  "amount-cur": "cur-eur",
+  "digital": false
+}
+----
+
+* Decimal with missing field: `"{\"decimal-int\": \"42\"}"`
++
+unescaped content:
++
+[source, json]
+----
+{"decimal-int": "42"}
+----
+
+* Decimal with `null` value: `"{\"decimal-int\": \"42\", \"decimal-frac\": null}"`
++
+unescaped content:
++
+[source, json]
+----
+{"decimal-int": "42", "decimal-frac": null}
+----
+
+* Decimal with missing outer braces: `"\"decimal-int\": \"42\", \"decimal-frac\": \"0\""`
++
+unescaped content:
++
+[source, json]
+----
+"decimal-int": "42", "decimal-frac": "0"
+----
+
+* Decimal with field name as JSON _keys_: `"{\"int\": \"42\", \"frac\": \"0\"}"`
++
```

Comment:
I don't understand what i meant by a JSON key.
This looks like "Decimal with incorrect field names"

Reply:
all italic words refer to JSON spec (mentioned above).